src/konflux-rpm-lockfile

```diff
@@ -0,0 +1,295 @@
+
+
+def write_hermeto_lockfile(arch_packages, repos):
+    """
+    Writes the hermeto lockfile structure.
+    """
+    arches = []
+    for arch_data in arch_packages:
+        arch = arch_data['arch']
+        pkgs = arch_data['packages']
+        formatted_packages = format_packages_with_repoid(pkgs, repos)
+        arches.append({
+            'arch': arch,
+            'packages': formatted_packages
+        })
+
+    lockfile = {
+        'lockfileVersion': 1,
+        "lockfileVendor": "redhat",
+        "arches": arches
+    }
+
+    return lockfile
+
+
+def merge_lockfiles(base_lockfile, next_lockfile, override=False):
+    """
+    Merges a lockfile into a base lockfile.
+
+    If is_override is True, it will only add packages to existing
+    architectures. Otherwise, it will add new architectures.
+    """
+    if not next_lockfile:
+        return base_lockfile
+
+    # Create a dictionary for base arches for easy lookup
+    base_arches = {arch['arch']: arch for arch in base_lockfile.get('arches', [])}
+
+    next_arches_list = next_lockfile.get('arches', [])
+    if not next_arches_list:
+        return base_lockfile
+
+    for next_arch_entry in next_arches_list:
+        if not isinstance(next_arch_entry, dict):
+            continue
+        arch = next_arch_entry.get('arch', None)
+        if not arch:
+            continue
+
+        next_packages = next_arch_entry.get('packages', [])
+        if arch in base_arches:
+            # Arch exists, merge packages
+            base_packages = base_arches[arch].get('packages', [])
+            base_packages += next_packages
+        elif not override:
+            # Arch is new and this is not an override, so add it
+            base_arches[arch] = next_arch_entry
+
+    # Reconstruct the arches list
+    base_lockfile['arches'] = list(base_arches.values())
+    return base_lockfile
+
+
+def query_packages_location(locks, repoquery_args):
+    """
+    Resolves packages URLs for a given architecture.
+    """
+    pkg_urls = []
+    if not locks:
+        return pkg_urls
+
+    locked_nevras = [f'{k}-{v}' for (k, v) in locks.items()]
+    cmd = ['dnf', 'repoquery'] + locked_nevras + repoquery_args
+    result = subprocess.check_output(cmd, text=True)
+
+    processed_urls = {}
+    for line in result.strip().split('\n'):
+        # ignore empty lines
+        if not line:
+            continue
+        parts = line.split(' ', 1)
+        # If more than 1 URL is returned we pick the first one
+        if len(parts) == 2:
+            name, url = parts
+            if name not in processed_urls:
+                processed_urls[name] = url
+    pkg_urls = list(processed_urls.values())
+    # sanity check all the locked packages got resolved
+    if len(pkg_urls) != len(locked_nevras):
+        print("Some packages from the lockfile could not be resolved. The rpm-ostree lockfile is probably out of date.")
+        sys.exit(1)
+
+    print(f"Done. Solved {len(pkg_urls)} packages.")
```

Comment:
```suggestion
    print(f"Done. Resolved location for {len(pkg_urls)} packages.")
```

Reply:
haha I knew `Solved` was technically incorrect but didn't change it. Thanks for the suggestion